jUnit Report Segfaults

Started seeing this on the branch builds 100% from https://openshift-gce-devel.appspot.com/build/origin-ci-test/logs/test_branch_origin_check/2084/

```
[INFO] jUnit XML report placed at _output/scripts/shell/artifacts/oscmd_report_Hr5GW.xml
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4d9147]
goroutine 1 [running]:
github.com/openshift/origin/tools/junitreport/pkg/parser/stack.(*testOutputParser).Parse(0xc420016b80, 0xc42001a700, 0x0, 0x5944c0, 0xc420016b80)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/tools/junitreport/pkg/parser/stack/parser.go:96 +0xc17
github.com/openshift/origin/tools/junitreport/pkg/cmd.(*JUnitReportOptions).Run(0xc420020480, 0x7ffd24fd949b, 0x6)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/tools/junitreport/pkg/cmd/junitreport.go:98 +0x1ef
main.main()
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/tools/junitreport/junitreport.go:162 +0x542
[ERROR] PID 3585: hack/lib/test/junit.sh:191: `[[ -n "${LOG_DIR:-}" && -d "${LOG_DIR-}" ]]` exited with status 2.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/lib/test/junit.sh:191: `[[ -n "${LOG_DIR:-}" && -d "${LOG_DIR-}" ]]`
[INFO] 		  2: hack/lib/test/junit.sh:167: os::test::junit::internal::generate_report
[INFO] 		  3: hack/test-cmd.sh:10: os::test::junit::generate_report
[INFO] 		  4: hack/test-cmd.sh:1: cleanup
[INFO]   Exiting with code 2.
```

/cc @smarterclayton @liggitt 
Have we changed the `os::cmd` output or parser in... forever? How is this happening only on branch builds and not on PRs?

--- a/hack/lib/init.sh
+++ b/hack/lib/init.sh
@@ -51,8 +51,8 @@ os::log::stacktrace::install
 os::util::environment::update_path_var
 
 if [[ -z "${OS_TMP_ENV_SET-}" ]]; then
-	if [[ "${BASH_SOURCE[0]}" =~ .*\.sh ]]; then
-		os::util::environment::setup_tmpdir_vars "$( basename "${BASH_SOURCE[0]}" ".sh" )"
+	if [[ "${0}" =~ .*\.sh ]]; then
+		os::util::environment::setup_tmpdir_vars "$( basename "${0}" ".sh" )"
 	else
 		os::util::environment::setup_tmpdir_vars "shell"
 	fi

